### PR TITLE
Remove Auxiliary State

### PR DESCRIPTION
**Pull request by mdpye**
We do not require to track last change or version in the currently
agreed design, instead differences are calculated closer to the
customer.

### CODE REVIEW

File: chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/state/JoinedRoomsReceivedReducer.kt [hunk 1]
```diff
@@ -9,5 +9,7 @@ internal val joinedRoomsReceivedReducer =
 
         val joinedRooms = action.rooms.map { it.id to it }.toMap()
         val joinedRoomsState = JoinedRoomsState(joinedRooms, action.unreadCounts)
-        state.with(joinedRoomsState, state.auxiliaryState.with(action))
-}
+        state.with(
+            joinedRoomsState = joinedRoomsState
+        )
```
Comment (marekoid) [inline chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/state/JoinedRoomsReceivedReducer.kt]: minor but if it fits 1 line it's usually more legible to not make it 3 lines (as it used to be here), also named param here is just noise

Comment (mdpye) [reply chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/state/JoinedRoomsReceivedReducer.kt]: It matches all the other invocations in other files, which makes it easier to read. I added the names parameter as we expect more in the (very near) future, right?

File: chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/state/RoomUpdatedReducer.kt [hunk 2]
```diff
@@ -6,13 +6,12 @@ import org.reduxkotlin.reducerForActionType
...
+        checkNotNull(state.joinedRoomsState)
 
-            state.with(
-                joinedRoomsState = JoinedRoomsState(
-                    state.joinedRoomsState.rooms.plus(action.room.id to action.room),
-                    state.joinedRoomsState.unreadCounts
-                ),
-                auxiliaryState = state.auxiliaryState.with(action)
+        state.with(
+            joinedRoomsState = JoinedRoomsState(
```
Comment (marekoid) [inline chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/state/RoomUpdatedReducer.kt]: named param does not help with signal-to-noise ratio here

**Comment by mdpye** [comment]
> - Coverage     37.26%   35.40%   -1.86%

File: chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/state/JoinedRoomsReceivedReducer.kt (hunk 1 above)
Comment (mdpye) [reply chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/state/JoinedRoomsReceivedReducer.kt]: Improved.

**Comment by marekoid** [comment]
> I don't really understand this, as I deleted code, but didn't have to change or delete any tests.

The code you deleted was being covered, though no assertions were being made. When removing covered code, the covered/uncovered ratio increases.

**Comment by marekoid** [comment]
We made decision to not keep making assertions around AuxiliaryState as it was trivial code and correctness check was supported by the help of the compiler so it felt like clearly negative ROI/maintanance.

**Pull request merged**